### PR TITLE
Support File Formats

### LINKED ISSUE #317: Support Configuration File Formats
ESLint 1.10 deprecates `.eslintrc` and recommends using `.eslintrc.js`, `.eslint.yaml`, `eslint.yml`, or `.eslint.json`.

Reference http://eslint.org/blog/2015/11/eslint-v1.10.0-released/#configuration-file-formats

### PR DESCRIPTION
**Pull request by ChristianMurphy**
resolves #317

### CODE REVIEW

**Comment by steelbrain** [comment]
@ChristianMurphy Be advised, this PR has got merge conflicts

**Comment by ChristianMurphy** [comment]
Resolved conflicts

File: lib/worker.js [hunk 1]
```diff
@@ -15,15 +15,18 @@ const eslintPathLocal = Path.join(FS.realpathSync(Path.join(__dirname, '..')), '
 let eslint = null
 let prefixPath = null
 
-function find(startDir, name) {
-  let filePath
+function find(startDir, names) {
```
Comment (steelbrain) [inline lib/worker.js]: This function is used in several places, not all of them pass array, so you should add something like
```js
if (typeof names === 'string') {
  names = [names]
}
```

Comment (ChristianMurphy) [inline lib/worker.js]: Updated to test for type `string`.
Added a new `localNames` variable to avoid param reassign.

**Comment by steelbrain** [comment]
Now all we need is a CHANGELOG entry

**Comment by ChristianMurphy** [comment]
Added CHANGELOG entry

**Comment by steelbrain** [comment]
CircleCI seems to be playing dumb, anyways, LGTM. Thanks!

**Comment by Arcanemagus** [comment]
When it fails at an intermediate step like that you should just restart the build.

File: lib/worker.js [hunk 2]
```diff
@@ -15,15 +15,24 @@ const eslintPathLocal = Path.join(FS.realpathSync(Path.join(__dirname, '..')), '
...
+  let localNames;
+  if (typeof name === 'string') {
```
Comment (tschaub) [inline lib/worker.js]: This should be `names`.  See #325 for a proposed fix.

**Pull request merged**